spaceMissionManager-service/src/main/java/cz/muni/fi/services/impl/MissionServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+        }
+        mission.setEndDate(endDate);
+        mission.setActive(false);
+        mission.setResult("Mission{" +
+                        "astronauts=" + mission.getAstronauts() +
+                        ", spacecrafts=" + mission.getSpacecrafts() +
+                        ", name='" + mission.getName() + '\'' +
+                        ", destination='" + mission.getDestination() + '\'' +
+                        ", eta=" + mission.getEta() +
+                        ", missionDescription='" + mission.getMissionDescription() + '\'' +
+                        ", endDate=" + mission.getEndDate() +
+                    '}');
+        try {
+            missionDao.updateMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not archive mission.", e);
+        }
+    }
+
+    @Override
+    public void createMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.createMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not create mission.", e);
+        }
+    }
+
+    @Override
+    public void cancelMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        try {
+            missionDao.cancelMission(mission);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not cancel mission.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions() throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions());
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public List<Mission> findAllMissions(boolean active) throws DataAccessException {
+        try {
+            return Collections.unmodifiableList(missionDao.findAllMissions(active));
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find missions.", e);
+        }
+    }
+
+    @Override
+    public Mission findMissionById(Long id) throws DataAccessException {
+        if (id == null){
+            throw new IllegalArgumentException("Id must not be null.");
+        }
+        try {
+            return missionDao.findMissionById(id);
+        } catch (Throwable e) {
+            throw new ServiceDataAccessException("Could not find mission.", e);
+        }
+    }
+
+    @Override
+    public void updateMission(Mission mission) throws DataAccessException {
+        if (mission == null){
+            throw new IllegalArgumentException("Mission must not be null.");
+        }
+        boolean missionActive;
+        try {
```

Comment:
try not to use catch when you can expect NullPointerException I suggest using simple if and 1 check if the method findMissionById returned null or a Mission

Reply:
Should be fixed now, thanks